src/main/java/sparksoniq/jsoniq/item/Base64BinaryItem.java

```diff
@@ -0,0 +1,167 @@
+package sparksoniq.jsoniq.item;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import org.rumbledb.api.Item;
+import sparksoniq.exceptions.IteratorFlowException;
+import sparksoniq.exceptions.UnexpectedTypeException;
+import sparksoniq.jsoniq.compiler.translator.expr.operational.base.OperationalExpressionBase;
+import sparksoniq.jsoniq.runtime.metadata.IteratorMetadata;
+import sparksoniq.semantics.types.AtomicTypes;
+import sparksoniq.semantics.types.ItemType;
+import sparksoniq.semantics.types.ItemTypes;
+
+import javax.xml.bind.DatatypeConverter;
+import java.util.Arrays;
+import java.util.regex.Pattern;
+
+public class Base64BinaryItem extends AtomicItem {
+
+    private static final String b04char = "([AQgw])";
```

Comment:
These seem to be manual computations related to base64. Base64 libraries already take care of such things.

Using an existing library (like org.apache.commons.codec.binary.Base64 ) takes care of everything with just encode and decode calls, including checking that the literal is well-formed, and in an optimized manner. You should have nothing to do yourself apart from calling the encoding and decoding functions of the library.

This manual processing may explain the issues in the tests.

Reply:
I completely agree with you that should be the case ... **but** if I try INVALID examples (defined [here](http://www.datypic.com/sc/xsd/t-xsd_base64Binary.html), and invalid in Zorba as well) like:
- base64Binary("FB8")
- base64Binary("==0F")
- base64Binary("Q===")

those get encoded to FB8, ==0F and Q=== respectively if I use `Base64.encodeBase64(base64BinaryString.getBytes())` or `DatatypeConverter.parseBase64Binary(base64BinaryString);` as well, without prior check on the matching pattern. 

That is why, both for hexBinary and base64Binary, I decided to implement this necessary check using the pattern defined in the [W3C documentation](https://www.w3.org/TR/xmlschema11-2/#base64Binary).

> This manual processing may explain the issues in the tests.

What are the issues in the tests?